lib/dialects/postgres/index.js

```diff
@@ -241,20 +241,20 @@ class Client_PG extends Client {
     // Purposely not putting timeout on `pg_cancel_backend` execution because erroring
     // early there would release the `connectionToKill` back to the pool with
     // a `KILL QUERY` command yet to finish.
-    const conn = await this.acquireConnection();
+    const conn = await this.acquireRawConnection();
 
     try {
       return await this._wrappedCancelQueryCall(conn, connectionToKill);
     } finally {
       // NOT returning this promise because we want to release the connection
       // in a non-blocking fashion
-      this.releaseConnection(conn);
+      this.destroyRawConnection(conn);
```

Comment:
this change could cause unhandledPromiseRejection if connection destroy fails, because releaseConnection was non-throwing.

Reply:
Would you recommend to catch and log here?

Reply:
We might also just `await` the promise. As it was for MySQL dialect.

Actually why `this.releaseConnection(conn)` was awaited for MySQL while not for Postres? I would expect it to be symetric.

Reply:
Hard to say. It originates back to `https://github.com/knex/knex/pull/2636/files` and was originally implemented that way, with no explanation for why non-blocking is preferred there.

Reply:
It looks like MySQL implementation also originally had un-awaited `releaseConnection`, and the `await` was added here: https://github.com/knex/knex/pull/3666/files#diff-c46d41926fdd59ada0e2bb2f34472b1295493d6dad75e336385ef4d22c28c32cR183

So I guess it should be awaited for Postgres implementation as well.

Reply:
makes sense!

Reply:
Added `await` in 51b5191

Reply:
btw does releaseConnection error should fail cancelQuery? before it doesnt

Reply:
@maximelkin IDK, I just think it should do the same for both dialects (MySQL and Postgress). In current version `cancelQuery` fails with `releaseConnection` error for Postgress, and ignores it for MySQL.

I guess it is like that because [this chage] was applied just to MySQL not to all dialects symmetrically.

[this chage]: https://github.com/knex/knex/pull/3666/files#diff-c46d41926fdd59ada0e2bb2f34472b1295493d6dad75e336385ef4d22c28c32cR183

Reply:
I mean, releaseConnection https://github.com/knex/knex/blob/9c390036dbd98b1e2c37411db0eabdcb1247e9a4/lib/client.js#L306
Doesn't throw, but  destroyRawConnection does

Reply:
@maximelkin Ok. I see. So what should I do with such exception?

Reply:
catch and log error, this will be backward compatible

Reply:
@maximelkin By logging the error you mean `this.logger`?

```js
try {
  return await this._wrappedCancelQueryCall(conn, connectionToKill);
} finally {
  this.destroyRawConnection(conn).catch((err) => {
    this.logger.warn(`Connection Error: ${err}`);
  })
}
```

Or something else?

What about MySQL? There I noticed that MySQL `destroyRawConnection` [catches errors itself](https://github.com/knex/knex/blob/master/lib/dialects/mysql/index.js#L73-L74):
```js
  async destroyRawConnection(connection) {
    try {
      const end = promisify((cb) => connection.end(cb));
      return await end();
    } catch (err) {
      connection.__knex__disposed = err;
    } finally {
      // see discussion https://github.com/knex/knex/pull/3483
      defer(() => connection.removeAllListeners());
    }
  }
```
So then I guess for MySQL it is OK as it is, isn't it?

Reply:
Maybe log __knex_disposed error?

Reply:
@maximelkin catched and logged in f93c03f.